### ISSUE TITLE
Laravel 5.2 middleware issue

### ISSUE THREAD

**Issue by opheliadesign** [GeneaLabs/laravel-caffeine#11]
Ran into this error after upgrading to Laravel 5.2:

```
Symfony\Component\Debug\Exception\FatalErrorException thrown with message "Interface 'Illuminate\Contracts\Routing\Middleware' not found"

Stacktrace:
#35 Symfony\Component\Debug\Exception\FatalErrorException in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/genealabs/laravel-caffeine/src/Http/Middleware/LaravelCaffeineDripMiddleware.php:7
#34 Symfony\Component\Debug\Exception\FatalErrorException:__construct in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:133
#33 Illuminate\Foundation\Bootstrap\HandleExceptions:fatalExceptionFromError in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:118
#32 Illuminate\Foundation\Bootstrap\HandleExceptions:handleShutdown in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:0
#31  in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/composer/ClassLoader.php:412
#30 Composer\Autoload\includeFile in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/composer/ClassLoader.php:301
#29 Composer\Autoload\ClassLoader:loadClass in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Container/Container.php:736
#28 spl_autoload_call in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Container/Container.php:736
#27 ReflectionClass:__construct in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Container/Container.php:736
#26 Illuminate\Container\Container:build in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Container/Container.php:631
#25 Illuminate\Container\Container:make in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:674
#24 Illuminate\Foundation\Application:make in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:123
#23 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php:64
#22 Illuminate\Foundation\Http\Middleware\VerifyCsrfToken:handle in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#21 call_user_func_array:{/home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#20 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php:49
#19 Illuminate\View\Middleware\ShareErrorsFromSession:handle in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#18 call_user_func_array:{/home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#17 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php:62
#16 Illuminate\Session\Middleware\StartSession:handle in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#15 call_user_func_array:{/home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#14 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php:37
#13 Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse:handle in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#12 call_user_func_array:{/home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#11 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php:59
#10 Illuminate\Cookie\Middleware\EncryptCookies:handle in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#9 call_user_func_array:{/home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#8 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php:44
#7 Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode:handle in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#6 call_user_func_array:{/home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:124
#5 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:103
#4 call_user_func:{/home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:103} in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:103
#3 Illuminate\Pipeline\Pipeline:then in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:132
#2 Illuminate\Foundation\Http\Kernel:sendRequestThroughRouter in /home/vagrant/PhpstormProjects/fvrs-version-2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:99
#1 Illuminate\Foundation\Http\Kernel:handle in /home/vagrant/PhpstormProjects/fvrs-version-2/public/index.php:54
#0 {main} in /home/vagrant/PhpstormProjects/fvrs-version-2/public/index.php:0
```

**Comment by mikebronner** [comment]
Thanks for posting. :) Haven't upgraded yet myself. Will look at this later today.

**Comment by lhdev** [comment]
I've confirmed this as well.  Please see here:
http://laravel.com/docs/5.2/upgrade#upgrade-5.2.0

Specifically the Deprecations sections relating to "The Illuminate\Contracts\Routing\Middleware contract has been deprecated."

**Comment by mikebronner** [comment]
Thanks guys for your patience! :) This is now fixed in version 0.3.0. Please update your composer.json as specified in the README (use version 0.3.0 and up for Laravel 5.2).
